src/main/kotlin/org/rust/lang/core/resolve/ref/RsMarcoItemReferenceContributor.kt

```diff
@@ -0,0 +1,24 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.lang.core.resolve.ref
+
+import com.intellij.patterns.PlatformPatterns
+import com.intellij.psi.*
+import com.intellij.util.ProcessingContext
+import org.rust.lang.core.psi.RsElementTypes.IDENTIFIER
+
+class RsMarcoItemReferenceContributor : PsiReferenceContributor() {
+    override fun registerReferenceProviders(registrar: PsiReferenceRegistrar) {
+        val pattern = PlatformPatterns.psiElement(IDENTIFIER)//never called
```

Comment:
I don't think that creating a reference will actually work here, even if we implement `ContributedReferenceHost`. Consider this (incomplete) code:

```
macro_rules foobar! { ... }
fn fooquux() { ... }

fn main() {
    let _ = foo
}
```

Here, we can't really say if `foo` is a macro refernce, or a "usual" function reference. However, completion contributor should still be able to provide completions.